File: 2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.cpp
class Solution {
    bool ispali(string s){
        int n=s.size();
        for(int i=0;i<(n/2)+1;i++) if(s[i]!=s[n-i-1]) return false;
        return true;
    }
public:
    string firstPalindrome(vector<string>& words) {
        for(string s:words) if(ispali(s)) return s;
        return "";
    }
};

File: 2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.java
class Solution {
    boolean ispal(String s){
         if(s== null || s.length() == 0){
            return false;
         }
        for(int i = 0; i < s.length()/2; i++){
            char start = s.charAt(i);
            char end = s.charAt(s.length() - 1 - i);

            if(start != end){
                return false;
            }
        }
        return true;
    }
    public String firstPalindrome(String[] words) {
        for(int i=0;i<words.length;i++) if(ispal(words[i])) return words[i];
        return "";
    }
}